include/bluetooth/scan.h

```diff
@@ -515,6 +515,38 @@ void bt_scan_filter_remove_all(void);
  */
 int bt_scan_params_set(struct bt_le_scan_param *scan_param);
 
+#if CONFIG_BT_SCAN_CONN_ATTEMPTS_FILTER
```

Comment:
Don't need ifdef around the declaration.

If we do this it won't show up in the documentation build.
The application will get a link error if the feature is not enabled.